crates/bls-gadgets/src/y_to_bit.rs

```diff
@@ -83,20 +83,49 @@ impl<P: Bls12Parameters> YToBitGadget<P> {
                 Err(SynthesisError::AssignmentMissing)
             }
         })?;
-        let y_eq_bit = Boolean::alloc(cs.ns(|| "alloc y eq bit"), || {
-            if pk.y.c0.get_value().is_some() {
-                let half = P::Fp::modulus_minus_one_div_two();
-                Ok(pk.y.c0.get_value().get()?.into_repr() == half)
-            } else {
-                Err(SynthesisError::AssignmentMissing)
-            }
-        })?;
+
+        let y_eq_bit =
+            Boolean::alloc(cs.ns(|| "alloc y eq bit"), || {
+                if pk.y.c1.get_value().is_some() {
+                    Ok(pk.y.c1.get_value().get()?.into_repr()
+                        == pk.y.c0.get_value().get()?.into_repr())
+                } else {
+                    Err(SynthesisError::AssignmentMissing)
+                }
+            })?;
```

Comment:
Also isn't the y_eq_bit supposed to check if y_c1 == half?

Reply:
Nope - that was a bug, see https://github.com/celo-org/bls-zexe/issues/149

Reply:
Fair enough. But shouldn't the check then be y_c1 == 0?

Reply:
But shouldn't it then check if y_c1 == 0? Or am I missing something here?

Reply:
Yes, indeed, realized it as well, I just pushed a fix.